### ISSUE TITLE
Improve nodetool getendpoints to return shard in addition to node

### ISSUE THREAD

**Issue by gnumoreno** [scylladb/scylla#7266]
Current state:
```
nodetool getendpoints -- keyspace table pk1:pk2:pk3
192.168.0.10
192.168.0.11
192.168.0.12
```

Desired:
```
nodetool getendpoints -- keyspace table pk1:pk2:pk3
192.168.0.10 10
192.168.0.11 10
192.168.0.12 10
```
10 above being the shard

**Comment by vladzcloudius** [comment]
I like this one - this in particular would allow mapping `large_partitions/rows` content to corresponding shard.

However the best would be to actually have the shard info in the `large_partition/row` to save the hassle.

@avikivity @slivne What do you think?

**Comment by dyasny** [comment]
I wonder if the API actually exposes it

**Comment by avikivity** [comment]
I don't think we can implement getendpoints because we don't know smp::count for other nodes.

We can add it to the local large* tables.

**Comment by gnumoreno** [comment]
I like the idea of having this on large* tables as well. Let's go with this idea since Avi and Vlad are in agreement.

**Comment by haaawk** [comment]
> I don't think we can implement getendpoints because we don't know smp::count for other nodes.
> 
> We can add it to the local large* tables.

smp::count is gossiped by each node and we know it. It's used in CDC

**Comment by dyasny** [comment]
@avikivity if we're going with having this information in a table, it would be an easy and extra nice UX to have the nodetool command output point the user towards that table

**Comment by slivne** [comment]
@haaawk
- can you please reference the CDC gossip info you mentioned above. 
- this will force using this only after CDC is GAd wwhich is ok for master.

**Comment by slivne** [comment]
@haaawk - ping

**Comment by slivne** [comment]
@haaawk / @kbr- ping

**Comment by haaawk** [comment]
@slivne https://github.com/scylladb/scylla/blob/master/gms/application_state.hh#L65-L66

**Comment by haaawk** [comment]
Also https://github.com/scylladb/scylla/blob/052a8d036d8c5ef73cb2ace333def29accde3634/cdc/generation.cc#L200

**Comment by YeminLi** [comment]
As a scylla user, I think something like this will be very useful when debugging hot partition/shard issues.
